src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateSet.YYYY) {
+    dateSet.YY = dateSet.YYYY.slice(2);
+  }
+
+  if (dateSet.YY < 30) {
```

Comment:
This check assumes that YY is already set, which might not be the case if the original format did not include YY. Additionally, this conversion should only occur if the 'toFormat' array specifies a YYYY, otherwise, it should not be executed.